gen/generator.toml

```
@@ -0,0 +1,29 @@
+[general]
+library_name = "libspatialindex_c"
+# jll_pkg_name = "LibSpatialIndex_jll"
+output_api_file_path = "../src/include/sidx_api.jl"
+output_common_file_path = "../src/include/sidx_config.jl"
+print_using_CEnum = true
+use_julia_native_enum_type = true
+output_ignorelist = ["SIDX_C_API", "SIDX_THREAD", "STRDUP"]
+use_deterministic_symbol = true
+smart_de_anonymize = true
+extract_c_comment_style = "doxygen"
+fold_single_line_comment = true
+struct_field_comment_style = "outofline"
+enumerator_comment_style = "outofline"
+show_c_function_prototype = false
+
+[codegen]
+is_function_strictly_typed = false
```

May I suggest using ccall macros? I just find them so much more readable.

```suggestion
use_ccall_macro = true
is_function_strictly_typed = false
```

---

Agreed!